abi/codegen_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package abi
+
+import (
+	"encoding/json"
+	"go/format"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenerateAllStructs(t *testing.T) {
+	require := require.New(t)
+
+	abi := mustJSONParse[VM](t, string(mustReadFile(t, "testdata/abi.json")))
+
+	code, err := GenerateGoStructs(abi, "abi")
+	require.NoError(err)
+
+	expected := mustReadFile(t, "mockabi.test.go")
+
+	formatted, err := format.Source(removeCommentLines(expected))
+	require.NoError(err)
+
+	require.Equal(string(formatted), code)
+}
+
+func removeCommentLines(input []byte) []byte {
```

Comment:
It's actually only for one particular test, so it doesn't matter much. This specific solution removes an extra \n, so it wouldn't work.

Reply:
Could we just change this to remove header and strip the first 3 lines of code so that it does exactly what we need and it's clear that it's not a general solution?